### ISSUE TITLE
Hono router generic type-errors if it receives an `interface` instead of a `type`

### ISSUE THREAD

**Issue by scrawnyColeman** [honojs/hono#2981]
### What version of Hono are you using?

4.4.6

### What runtime/platform is your app running on?

Cloudflare Pages

### What steps can reproduce the bug?

Attempt to initialise `Hono` router with an `interface`. I have attempted to do so with cloudflare pages. The interface generated by wrangler looks as follows:
```
type Env = {
	DB: D1Database;
};
```

### What is the expected behavior?

There should be no `ts-error`. This is an example of the `type` syntax

<img width="580" alt="image" src="https://github.com/honojs/hono/assets/43850448/1209c09f-9555-4670-9762-abb6bdc17d03">

### What do you see instead?

<img width="568" alt="image" src="https://github.com/honojs/hono/assets/43850448/1f67b0fe-0424-463c-a7d5-e017b73d70ac">

The following type-error appears:
```
Type '{ Bindings: Env; }' does not satisfy the constraint 'Env'.
  Types of property 'Bindings' are incompatible.
    Type 'Env' is not assignable to type 'Bindings'.
      Index signature for type 'string' is missing in type 'Env'.
```

**Comment by scrawnyColeman** [comment]
cc @RamIdeas

**Comment by yusukebe** [comment]
Hi @scrawnyColeman 

This is not a real bug but a known issue related to TypeScript's limitations. Currently, we only have to write it as `type`. However, I also think it should be written with `interface`. So, let this issue be left as an enhancement.

**Comment by sameerdewan** [comment]
This is a bug as it introduces incompatibility with the produced types from the command `wrangler types` or `npm run cf-typegen`. How can we get around this? Currently, the only seen way is to manually override interface -> type after *every* run of `wrangler types` or `npm run cf-typegen`

**Comment by ottomated** [comment]
Fixed by #3136

**Comment by yusukebe** [comment]
@ottomated Thanks!

@sameerdewan Thanks to @ottomated , it was fixed. You can now use an `interface` as a `Bindings`.
